subsys/bluetooth/host/l2cap.c

```diff
@@ -1996,16 +1947,35 @@ static int l2cap_chan_le_send(struct bt_l2cap_le_chan *ch,
 
 	if (!test_and_dec(&ch->tx.credits)) {
 		LOG_DBG("No credits to transmit packet");
+
 		return -EAGAIN;
 	}
 
 	/* Save state so it can be restored if we failed to send */
 	net_buf_simple_save(&buf->b, &state);
 
-	seg = l2cap_chan_create_seg(ch, buf, sdu_hdr_len);
-	if (!seg) {
-		atomic_inc(&ch->tx.credits);
-		return -EAGAIN;
+	if (net_buf_frags_len(buf) <= ch->tx.mps) {
+		LOG_DBG("len < MPS, not allocating seg for %p", buf);
+		seg = net_buf_ref(buf);
```

Comment:
This look suspicious to me. I think we should move the reference from `buf` to `seg`, not copy it. Can you double-check this?

Reply:
let's look at it IRL today

Reply:
Looks ok. `buf` is not moving a reference. It's a borrow.